File: p003/p003.py
# Henrik Grønbech; https://projecteuler.net/problem=3
import time

def main():
    n = 600851475143
    
    biggest_factor = 2
    while n > 1:
        for i in range(biggest_factor, n+1, biggest_factor % 2 + 1):
            if n % i == 0:
                n = n // i
                biggest_factor = i
                break
    
    print(biggest_factor)


if __name__ == '__main__':
    main()

File: p003/p003.jl
# Henrik Grønbech; https://projecteuler.net/problem=3

function main()
    n = 600851475143;
    biggest_factor = 2;

    while n > 1
        for i = biggest_factor:(mod(biggest_factor, 2)+1):n
            if mod(n, i) == 0
                n = n ÷ i;
                biggest_factor = i;
                break
            end
        end
    end

    println(biggest_factor);
end

main()
